### PR TITLE
Craig-927: added ClassicVlanForm and story

### PR DESCRIPTION
**Pull request by Pseusco**

### CODE REVIEW

File: src/components/forms/ClassicVlanForm.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+class ClassicVlanForm extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { ...this.props.data};
+        this.handleInputChange = this.handleInputChange.bind(this);
+        buildFormDefaultInputMethods(this);
+        buildFormFunctions(this);
+    }
+
+      /**
```
Comment (evpirnia) [inline src/components/forms/ClassicVlanForm.js:21]: format please with `npm run pretty`

**Review by evpirnia — CHANGES_REQUESTED** [review]

File: storybook/src/stories/Forms/ClassicVlanForm.stories.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  title: "Components/Forms/ClassicVlanForm",
+  argTypes: {
+    data: {
+      summary: "An object that contains the data for the form.",
+      type: { required: false }, // required prop or not
+      control: "none",
+    },
+    ["data.name"]: {
+      description: "A string value for the name of the VLAN",
+      type: { required: false },
```
Comment (Ay1man2) [inline storybook/src/stories/Forms/ClassicVlanForm.stories.js:16]: change `required: true`

File: storybook/src/stories/Forms/ClassicVlanForm.stories.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      control: "none",
+    },
+    ["data.type"]: {
+      description: "A string value for the Vlan type, public or private",
+      type: { required: false },
```
Comment (Ay1man2) [inline storybook/src/stories/Forms/ClassicVlanForm.stories.js:21]: change `required: true`

File: storybook/src/stories/Forms/ClassicVlanForm.stories.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      control: "none",
+    },
+    ["data.datacenter"]: {
+      description: "A string value for the VLAN datacenter",
+      type: { required: false },
```
Comment (Ay1man2) [inline storybook/src/stories/Forms/ClassicVlanForm.stories.js:26]: change `required: true`

**Comment by jvallexm** [comment]
needs merge, bump version to 1.2.11

File: src/components/forms/ClassicVlanForm.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            componentProps={this.props}
+            hideHelperText
+            invalid={this.props.invalidCallback(this.state, this.props)}
+            invalidText={this.props.invalidTextCallback(this.state, this.props)}
+          />
+          <IcseSelect
+            labelText={"VLAN Type"}
+            name={"type"}
+            formName={this.props.data.name + "-type"}
+            groups={this.props.vlanTypes}
```
Comment (jvallexm) [inline src/components/forms/ClassicVlanForm.js:45]: this can always be `PUBLIC` and `PRIVATE`

should be title case when rendered, all caps in state

**Review by jvallexm — CHANGES_REQUESTED** [review]

**Review by jvallexm — APPROVED** [review]

File: src/components/forms/ClassicVlanForm.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            componentProps={this.props}
+            hideHelperText
+            invalid={this.props.invalidCallback(this.state, this.props)}
+            invalidText={this.props.invalidTextCallback(this.state, this.props)}
+          />
+          <IcseSelect
+            labelText={"VLAN Type"}
+            name={"type"}
+            formName={this.props.data.name + "-type"}
+            groups={["PUBLIC", "PRIVATE"].map(vlanTypeTitleCase)}
```
Comment (jvallexm) [inline src/components/forms/ClassicVlanForm.js:55]: Just hardcode here, it'll always be uppercase in state

**Review by jvallexm — CHANGES_REQUESTED** [review]

File: src/components/forms/ClassicVlanForm.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+            value={vlanTypeTitleCase(this.state.type)}
```
Comment (jvallexm) [inline src/components/forms/ClassicVlanForm.js:56]: `titleCase(this.state.typ`

File: src/components/forms/ClassicVlanForm.js (hunk 6 above)
Comment (jvallexm) [reply src/components/forms/ClassicVlanForm.js:55]: `["Public", "Private"]`

**Review by jvallexm — APPROVED** [review]

**Review by evpirnia — APPROVED** [review]

**Review by Ay1man2 — APPROVED** [review]

**Pull request merged**